applications/GeoMechanicsApplication/custom_utilities/check_utilities.cpp

```diff
@@ -25,28 +27,48 @@ void CheckUtilities::CheckDomainSize(double DomainSize, std::size_t Id, const st
         << min_domain_size << " for element " << Id << std::endl;
 }
 
-void CheckUtilities::CheckNodalVariables(const Geometry<Node>&            rGeometry,
-                                         const std::vector<VariableData>& rVariables)
+void CheckUtilities::CheckHasNodalSolutionStepData(const Geometry<Node>& rGeometry, 
+    const Geo::ConstVariableDataRefs& rVariableRefs)
 {
-    for (std::size_t i = 0; i < rGeometry.PointsNumber(); ++i) {
-		for (const auto & r_variable : rVariables){
-           if (!rGeometry[i].SolutionStepsDataHas(r_variable))
-               KRATOS_ERROR << "Missing variable " << r_variable.Name() << " on node "
-                            << rGeometry[i].Id() << std::endl;
-		}
-	}
+    for (const auto& r_variable_ref : rVariableRefs) {
+        std::vector<std::size_t> missing_node_ids;
+        for (const auto& node : rGeometry) {
+            if (!node.SolutionStepsDataHas(r_variable_ref.get())) {
+                missing_node_ids.push_back(node.Id());
+            }
+        }
+        if (!missing_node_ids.empty())
+           KRATOS_ERROR << "Missing variable " << r_variable_ref.get().Name() << " on nodes "
+                         << PrintVectorContent(missing_node_ids) << std::endl;
+    }
 }
 
-void CheckUtilities::CheckNodalDof(const Geometry<Node>&            rGeometry,
-                                   const std::vector<VariableData>& rVariables)
+void CheckUtilities::CheckHasDofs(const Geometry<Node>& rGeometry, const Geo::ConstVariableDataRefs& rVariableRefs)
 {
-    for (std::size_t i = 0; i < rGeometry.PointsNumber(); ++i) {
-        for (const auto& r_variable : rVariables) {
-            if (!rGeometry[i].HasDofFor(r_variable))
-                KRATOS_ERROR << "Missing the DoF for the variable " << r_variable.Name() << " on node "
-                             << rGeometry[i].Id() << std::endl;
+    for (const auto& r_variable_ref : rVariableRefs) {
+        std::vector<std::size_t> missing_node_ids;
+        for (const auto& node : rGeometry) {
+            if (!node.HasDofFor(r_variable_ref.get())) {
+                missing_node_ids.push_back(node.Id());
+            }
         }
+        
+        if (!missing_node_ids.empty())
+            KRATOS_ERROR << "Missing the DoF for the variable " << r_variable_ref.get().Name()
+                             << " on nodes " << PrintVectorContent(missing_node_ids) << std::endl;
     }
 }
 
+    std::string CheckUtilities::PrintVectorContent(const std::vector<size_t>& rVector)
+{
+    std::ostringstream oss;
+    for (const auto& r_value : rVector)
+        oss << r_value << " ";
+
+    std::string output = oss.str();
+    output.pop_back();
```

Comment:
Nitpicking: trying to remove the last character from an empty string will fail. This happens when `rVector` is empty. To make the implementation more robust, you could add the following check:
```suggestion
    if (!output.empty()) output.pop_back();
```
(I understand that at present, this function is only being called when `rVector` is not empty. But nothing stops a developer from calling this function with an empty vector. Being able to handle that case as well seems like a good idea to me (to avoid unexpected surprises)).

Reply:
Hi Anne, thank you for the suggestion. Yes, it is safer.